From 77c2ef935c208a5576ba3494fd9a42787772680e Mon Sep 17 00:00:00 2001
From: Subrata Chatterjee <schatterjee@habana.ai>
Date: Tue, 03 Jan 2023 13:29:47 +0200
Subject: [PATCH] [SW-103768] gaudi2: bist: remove redundant steps

Incase BIRA fails, BISR and BIST are performed for
a ring after BIRA in sequence. So id_select and ring_select
steps do not need to be done for all the operations.
This patch modifies common wrapper to call id_select and
ring_select functions only for BIRA.

Change-Id: Ie063036af6d1645c558ce506c796cd7bacfecb06
Signed-off-by: Subrata Chatterjee <schatterjee@habana.ai>
---

diff --git a/arch/gaudi2/arc/asic.c b/arch/gaudi2/arc/asic.c
index 3163731..71e0b85 100644
--- a/arch/gaudi2/arc/asic.c
+++ b/arch/gaudi2/arc/asic.c
@@ -248,8 +248,10 @@
 {
 	int ret;
 
-	gaudi2_sram_id_sel();
-	gaudi2_sram_ring_sel(dfx_value_table[dfx], ring);
+	if (op == SRAM_SELF_REPAIR_OP_BIRA) {
+		gaudi2_sram_id_sel();
+		gaudi2_sram_ring_sel(dfx_value_table[dfx], ring);
+	}
 
 	ret = gaudi2_sram_ring_pattern(op);
 
